### ISSUE TITLE
Integrating  with other widgets such as react-select or react-tags #396

### ISSUE THREAD

**Issue by atkawa7** [mozilla-services/react-jsonschema-form#399]
How do you make sure the rest of fields render with default widgets and then the custom field with your custom widget.

**Comment by revolunet** [comment]
not sure i understand correctly the question but here's an example where i use some custom widget only for a given field (description)

```js
import Form from 'react-jsonschema-form';

// dummy widget example, just with value & onChange
const MyCustomWidget = ({ formData, onChange }) => <div>{ formData }</div>;

const formProps = {
  fields: {
    description: MyCustomWidget
  },
  schema: {
    title: 'Description',
    type: 'object',
    required: ['description'],
    properties: {
      description: {type: 'string', title: 'Text'},
      notes: {type: 'string', title: 'Notes'},
    }
  },
  uiSchema: {
    description: {
      'ui:field': 'description',
      'ui:options': {
        size: 'big'
      }
    }
  },
  formData: {
    description: "some description",
  }
};

const Demo2 = () => (
  <div>
    <Form { ...formProps } />
  </div>
)
```

**Comment by atkawa7** [comment]
I am trying to create a list of genres using reactinput tags

here  is what I have but I am still getting error 

```
import React from 'react';
import Form from 'react-jsonschema-form';
import SchemaTagField from './SchemaTagField';

const schema = {
    type: 'object',
    required: [
        'name', 'genres'
    ],
    properties: {
        name: {
            type: 'string'
        },
        genres: {
            type: 'array',
            items: {
                type: 'string',
                enum: [],
                enumNames: []
            }
        }

    }
};

const fields = {
    items: SchemaTagField
};
const uiSchema = {
    genres: {
        'ui:field': 'items'
    }
};

export default class Test extends React.Component {
    onChange(event) {
        console.log(event);
    }
    render() {
        return (
            <div className='col-md-4 col-md-offset-4'>
                <Form schema={schema} uiSchema={uiSchema} fields={fields}/></div>
        );

import React from 'react';
import TagsInput from 'react-tagsinput';
import '../css/tags.css';

export default class SchemaTagField extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            tags: []
        }
    }

    handleChange(tags) {
        this.setState({
            tags
        }, function() {
            var obj = {
                [this.props.uiSchema['ui:field']]: this.state.tags
            };
            // console.log(obj);
            this.props.onChange(obj);
        }.bind(this));
    }

    render() {
        const inputProps = {
            className: 'react-tagsinput-input',
            placeholder: `Add ${ [this.props.uiSchema['ui:field']]}`
        };
        return (<TagsInput value={this.state.tags} inputProps={inputProps} onChange={this.handleChange.bind(this)}/>);
    }
}
```

**Comment by atkawa7** [comment]
I am getting this error `Cannot read property 'items' of undefined`

**Comment by n1k0** [comment]
Why did you recreate an issue I've closed? (#396)

Don't do that. An issue teacker isn't for support requests. Keep commenting on #396 but don't recreate new issues about this, thanks.
